mailpoet/assets/js/src/segments/dynamic/list/bulk-actions.tsx

```diff
@@ -0,0 +1,202 @@
+import { useState } from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { SelectControl } from '@woocommerce/components';
+import { Button, Modal } from '@wordpress/components';
+import { dispatch, select } from '@wordpress/data';
+import { MailPoet } from '../../../mailpoet';
+import { DynamicSegment } from '../types';
+import { storeName } from '../store';
+
+type ConfirmModalProps = {
+  onClose: () => void;
+  onConfirm: () => void;
+  confirmText: string;
+  title: string;
+  message: string | JSX.Element;
+};
+function ConfirmModal({
+  onClose,
+  onConfirm,
+  confirmText,
+  title,
+  message,
+}: ConfirmModalProps): JSX.Element {
+  return (
+    <Modal title={title} onRequestClose={onClose}>
+      <p>{message}</p>
+      <Button variant="primary" onClick={onConfirm}>
+        {confirmText}
+      </Button>
+      <Button variant="tertiary" onClick={onClose}>
+        {__('Cancel', 'mailpoet')}
+      </Button>
+    </Modal>
+  );
+}
+
+async function bulkAction(action: string, segments: DynamicSegment[]) {
+  void MailPoet.Ajax.post({
+    api_version: 'v1',
+    endpoint: 'dynamic_segments',
+    action: 'bulk_action',
+    data: {
+      action,
+      listing: {
+        selection: segments.map((segment) => segment.id),
+      },
+    },
+  }).then(() => {
+    void dispatch(storeName).loadDynamicSegments();
+  });
```

Comment:
We should probably prefer `async`/`await`. Here it would have the side effect of revealing a potential request error here and not somewhere else. And yeah, errors are probably not handled here, but maybe they don't need to be, not sure.